packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -58,19 +67,73 @@ export async function mapETHWithdrawalToL2ToL1EventResult(
   }
 }
 
+function getFirstExecutableBlockFromCache({
+  event,
+  l2ChainID
+}: {
+  event: L2ToL1EventResult
+  l2ChainID: number
+}) {
+  const cacheKey = getMessagesCacheKey({ event, l2ChainId: l2ChainID })
+  const firstExecutableBlockCache = JSON.parse(
+    localStorage.getItem(firstExecutableBlockLocalStorageKey) || '{}'
+  )
+  const result = firstExecutableBlockCache[cacheKey]
+
+  if (!result) {
+    return undefined
```

Comment:
I think we can default to `-1` here, which is cool because it's a `number`, and it will work well with the calculation, so we don't have to do `if (firstExecutableBlockFromCache) { ... }`

```suggestion
    return -1
```

Reply:
done